apps/pipelines/tests/test_runnable_builder.py

```diff
@@ -16,76 +19,83 @@ def provider():
-                            "type": "RenderTemplate",
-                            "params": {
-                                "template_string": "<b>{{ summary }}</b>",
-                            },
+                        "label": "Create the report",
+                        "type": "CreateReport",
+                        "params": {
+                            "llm_provider_id": provider.id,
+                            "llm_model": "fake-model",
+                            "prompt": """Make a summary of the following text: {input}.
+                                Output it as JSON with a single key called 'summary' with the summary.""",
                         },
-                        "id": "template",
                     },
-                    {
-                        "data": {
-                            "id": "email",
-                            "label": "Send an email",
-                            "type": "SendEmail",
-                            "params": {
-                                "recipient_list": "test@example.com",
-                                "subject": "This is an interesting email",
-                            },
+                    "id": "report",
+                },
+                {
+                    "data": {
+                        "id": "template",
+                        "label": "render a template",
+                        "type": "RenderTemplate",
+                        "params": {
+                            "template_string": "<b>{{ summary }}</b>",
                         },
+                    },
+                    "id": "template",
+                },
+                {
+                    "data": {
                         "id": "email",
+                        "label": "Send an email",
+                        "type": "SendEmail",
+                        "params": {
+                            "recipient_list": "test@example.com",
+                            "subject": "This is an interesting email",
+                        },
                     },
-                ],
-                "id": 1,
-                "name": "New Pipeline",
-            }
-        )
+                    "id": "email",
+                },
+            ],
+            "id": 1,
+            "name": "New Pipeline",
+        }
+    )
 
-    runnable.invoke(PipelineState(messages=["Ice is not a liquid. When it is melted it turns into water."]))
+    state = PipelineState(
+        messages=["Ice is not a liquid. When it is melted it turns into water."], experiment_session_id=1
+    )
+    runnable.invoke(state)
     assert len(mail.outbox) == 1
     assert mail.outbox[0].subject == "This is an interesting email"
     assert mail.outbox[0].to == ["test@example.com"]
 
 
 @override_settings(CELERY_TASK_ALWAYS_EAGER=True)
-def test_send_email():
+@mock.patch("apps.pipelines.nodes.base.PipelineNode.logger")
+def test_send_email(logger):
```

Comment:
tip: If you don't need access to the mock object you can prevent having to add the argument by adding `new=Mock()` to the patch (kwarg not technically required since it's the first one):

```
@mock.patch("apps.pipelines.nodes.base.PipelineNode.logger", mock.Mock())
def test_send_email():
```

Reply:
https://github.com/dimagi/open-chat-studio/pull/474/commits/65986a3509e201feba514bac702e4d468f5e1d66